ambry-api/src/main/java/com/github/ambry/server/AccountStatsStore.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.server;
+
+import java.util.Map;
+
+
+/**
+ * The interface that stores and fetches account stats, aggregated account stats.
+ */
+public interface AccountStatsStore {
+  /**
+   * Store stats in the {@link StatsWrapper}. The StatsWrapper's {@link StatsSnapshot} should include partition, account
+   * and container. This method will be used for individual ambry server to store local stats.
+   * @param statsWrapper The {@link StatsWrapper} that contains stats and other metadata.
+   */
+  void storeStats(StatsWrapper statsWrapper);
+
+  /**
+   * Store aggregated stats in the {@link StatsSnapshot}. The StatsSnapshot should include account and container. This
+   * method will be used for aggregation task to store aggregated stats.
+   * @param snapshot The {@link StatsSnapshot} that contains aggregated container usage.
+   * @throws Exception
+   */
+  void storeAggregatedStats(StatsSnapshot snapshot) throws Exception;
+
+  /**
+   * Return individual ambry server's stats for the given {@code clusterName} and {@code hostname}. This is the stats
+   * stored by method {@link #storeStats}.
+   * @param clusterName The clusterName.
+   * @param hostname The hostname.
+   * @return {@link StatsWrapper} of given {@code hostname} in the {@code clusterName}.
+   * @throws Exception
+   */
+  StatsWrapper queryStatsOf(String clusterName, String hostname) throws Exception;
+
+  /**
+   * Return the aggregated stats for the given {@code clusterName}. This is the stats stored by method {@link #storeAggregatedStats}.
+   * Since the {@link StatsSnapshot} passed to {@link #storeAggregatedStats} only have account and container stats, this
+   * method returns a map back to the caller. The key of the outer map is the account id in string format, and the key of
+   * the inner map is the container id in string format, the value of the inner map is the storage usage for this container.
```

Comment:
why are the account/container IDs in string format instead of Integer? Also, I remember their being another format where the stats report is grouped by partition class. Is that handled somewhere else?

Reply:
I think it's because  the `subMap` in `StatsSnapshot` uses String type key (account/container and partition Id are converted to String when doing aggregation.)

Talked with Justin offline, in the interest of time, partition class stats report won't be migrated this quarter. We probably will take this up next year.

Reply:
Second Yingyi's comments. The partition class stats will be handled next quarter.